kafka_consumer/datadog_checks/kafka_consumer/new_kafka_consumer.py

```diff
@@ -101,14 +100,18 @@ def _load_broker_timestamps(self):
         """Loads broker timestamps from persistent cache."""
         self._broker_timestamps = defaultdict(dict)
         try:
-            for topic_partition, content in json.loads(read_persistent_cache(self._broker_timestamp_cache_key)).items():
+            json_cache = self._read_persistent_cache()
+            for topic_partition, content in json.loads(json_cache).items():
                 for offset, timestamp in content.items():
                     self._broker_timestamps[topic_partition][int(offset)] = timestamp
         except Exception as e:
             self.log.warning('Could not read broker timestamps from cache: %s', str(e))
 
+    def _read_persistent_cache(self):
+        return self._parent_check.read_persistent_cache(self._broker_timestamp_cache_key)
+
     def _save_broker_timestamps(self):
-        write_persistent_cache(self._broker_timestamp_cache_key, json.dumps(self._broker_timestamps))
+        self._parent_check.write_persistent_cache(self._broker_timestamp_cache_key, json.dumps(self._broker_timestamps))
```

Comment:
I needed to have `self._parent_check.write_persistent_cache` instead of doing `self.write_persistent_cache` because `NewKafkaConsumerCheck` does not actually inherit from `AgentCheck`. It instead takes a `parent_check` as a parameter.